### PR TITLE
BaSP-TG-99: Form - Super admin

### PR DESCRIPTION
**Pull request by Ansitox**
- Add a redux thunk to GET the entity by its ID
- Change some logic to make form work with new getByIdEmployee() thunk
- Implement react-hook-form
- Implement Joi validations and resolver

How to test QA:
1- Navigate to super admins entity in the navbar

2- Test create employee work as intended
2.1- Test new "reset" button, it should reset every field to blank
2.2- Test error messages (check ticket TG-99 Joi validations)

3- Test update employee work as intended
3.1- Test new "reset" button, it should reset every field to the original value
3.2- Test error messages (check ticket TG-99 Joi validations)

### CODE REVIEW

**Review by matescarabino — APPROVED** [review]
Looks fine

File: src/Components/SuperAdmins/Form/validations.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import Joi from 'joi';
+
+export const superAdminsValidationSchema = Joi.object({
+  name: Joi.string()
+    .min(3)
+    .pattern(/^[\p{L}]+$/u)
+    .required()
+    .messages({
```
Comment (ignacioaristo) [inline src/Components/SuperAdmins/Form/validations.js:8]: custom all the messages ( min, required)

File: src/Components/SuperAdmins/Form/validations.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      'string.pattern.base': 'Must contain only letters'
+    }),
+  lastName: Joi.string()
```
Comment (ignacioaristo) [inline src/Components/SuperAdmins/Form/validations.js:11]: same in all the messages

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/Components/SuperAdmins/Form/validations.js (hunk 2 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/Form/validations.js:13]: done

File: src/Components/SuperAdmins/Form/validations.js (hunk 1 above)
Comment (Ansitox) [reply src/Components/SuperAdmins/Form/validations.js:8]: done

**Review by CristianBucci — APPROVED** [review]
Everything looks good to me!

**Review by CristianBucci — APPROVED** [review]
Everything looks right!

**Review by NicoSan02 — APPROVED** [review]

**Pull request merged**